test/metabase/api/action_test.clj

```diff
@@ -53,57 +53,75 @@
-    :request-body (assoc (mt/mbql-query categories {:filter [:= $id 1]})
-                         :update_row {:name "updated_row"})
-    :expected     {:rows-updated [1]}}
-   {:action       "action/row/delete"
-    :request-body (mt/mbql-query categories {:filter [:= $id 1]})
-    :expected     {:rows-deleted [1]}}
-   {:action       "action/row/update"
-    :request-body (assoc (mt/mbql-query categories {:filter [:= $id 10]})
-                         :update_row {:name "new-category-name"})
-    :expected     {:rows-updated [1]}}])
+(defn- format-field-name
+  "Format `field-name` appropriately for the current driver (e.g. uppercase it if we're testing against H2)."
+  [field-name]
+  (keyword (mt/format-name (name field-name))))
+
+(defn- categories-row-count []
+  (first (mt/first-row (mt/run-mbql-query categories {:aggregation [[:count]]}))))
+
+(deftest create-test
+  (testing "POST /api/action/row/create"
+    (mt/test-drivers (mt/normal-drivers-with-feature :actions)
+      (actions.test-util/with-actions-test-data-and-actions-enabled
+        (let [response (mt/user-http-request :crowberto :post 200
+                                             "action/row/create"
+                                             (assoc (mt/mbql-query categories) :create-row {(format-field-name :name) "created_row"}))]
+          (is (schema= {:created-row {(format-field-name :id)   (s/eq 76)
+                                      (format-field-name :name) (s/eq "created_row")}}
+                       response)
+              "Create should return the entire row")
+          (let [created-id (get-in response [:created-row (format-field-name :id)])]
+            (is (= "created_row" (-> (mt/rows (mt/run-mbql-query categories {:filter [:= $id created-id]})) last last))
+                "The record at created-id should now have its name set to \"created_row\"")))))))
 
-(deftest happy-path-test
-  (testing "Make sure it's possible to use known actions end-to-end if preconditions are satisfied"
-    (actions.test-util/with-actions-test-data-and-actions-enabled
-      (doseq [{:keys [action request-body expected expect-fn]} (mock-requests)]
-        (testing action
-          (let [result (mt/user-http-request :crowberto :post 200 action request-body)]
-            (when expected (is (= expected result)))
-            (when expect-fn (expect-fn result))))))))
-
-(deftest create-update-delete-test
-  (testing "Make sure actions are acting on rows."
-    (actions.test-util/with-actions-test-data-and-actions-enabled
-      (let [[create update delete] (mock-requests)
-            {created-id :id :as created-row}
-            (:created-row (mt/user-http-request :crowberto :post 200 (:action create) (:request-body create)))]
-        (is (= [:id :name] (keys created-row))
-            "Create should return the entire row")
-        (is (= "created_row" (:name created-row))
-            "Create should return the correct value for name")
```

Comment:
`schema=` is perfectly suited for this type of thing and a lot easier to read, especially if the test fails